src/android/shakeDetector.java

```diff
@@ -0,0 +1,251 @@
+                Log.d(context, TAG, "no configuration found for event "+eventName+", skipping notification");
+                return;
+            }
+            JSONArray notifyConfigs = notifyConfigWrapper.getJSONArray(CONFIG_LIST_KEY);
+            JSONArray mutedConfigs = notifyConfigWrapper.optJSONArray(MUTED_LIST_KEY);
+
+            for(int i = 0; i < notifyConfigs.length(); i++) {
+                try {
+                    JSONObject currNotifyConfig = notifyConfigs.getJSONObject(i);
+                    int mutedIndex = findEntryWithId(mutedConfigs, currNotifyConfig.getLong(ID));
+                    if(mutedIndex == -1) {
+                        if (autogenData != null) { // we need to merge in the autogenerated data with any user data
+                            JSONObject currData = currNotifyConfig.optJSONObject("data");
+                            if (currData == null) {
+                                currData = new JSONObject();
+                                currNotifyConfig.put("data", currData);
+                            }
+                            mergeObjects(currData, autogenData);
+                        }
+                        Log.d(context, TAG, "generating notification for event "+eventName
+                                + " and id = " + currNotifyConfig.getLong(ID));
+                        Manager.getInstance(context).schedule(currNotifyConfig, TriggerReceiver.class);
+                    } else {
+                        Log.d(context, TAG, "notification for event "+eventName+" and id = "+currNotifyConfig.getLong(ID)
+                                +" muted, skip");
+                    }
+                } catch (Exception e) {
+                    Log.e(context, TAG, "Got error "+e.getMessage()+" while processing object "
+                            + notifyConfigs.getJSONObject(i) + " at index "+i);
+                }
+            }
+        } catch(JSONException e) {
+            Log.e(context, TAG, e.getMessage());
+            Log.e(context, TAG, e.toString());
+        }
+    }
+
+    private void mergeObjects(JSONObject existing, JSONObject autogen) throws JSONException {
+        JSONArray toBeCopiedKeys = autogen.names();
+        for(int j = 0; j < toBeCopiedKeys.length(); j++) {
+            String currKey = toBeCopiedKeys.getString(j);
+            existing.put(currKey, autogen.get(currKey));
+        }
+    }
+
+    private int findEntryWithId(JSONArray array, long id) throws JSONException {
+        if (array == null) {
+            return -1;
+        }
+        for (int i = 0; i < array.length(); i++) {
+            JSONObject currCheckedObject = array.getJSONObject(i);
+            if (currCheckedObject.getLong(ID) == id) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private void storeDataAndNotify(Context context, float[] accelerations, long currTime, float combinedAccel) throws JSONException{
```

Comment:
Although the name of the method is `storeDataAndNotify`, it doesn't appear to actually store any data. But I though that @ariasfelipe saved to the usercache because he used the data to get initial results....

Reply:
I guess he just used the logs, not the database...